docs/release-notes/flink-1.12.md

```diff
@@ -0,0 +1,169 @@
+
+- Deprecate `StreamExecutionEnvironment.setStreamTimeCharacteristic()` and `TimeCharacteristic` [FLINK-19319](https://issues.apache.org/jira/browse/FLINK-19319)
+
+  In Flink 1.12 the default stream time characteristic has been changed to `EventTime`, thus you don't need to call this method for enabling event-time support anymore. Explicitly using processing-time windows and timers works in event-time mode. If you need to disable watermarks, please use `ExecutionConfig.setAutoWatermarkInterval(long)`. If you are using `IngestionTime`, please manually set an appropriate `WatermarkStrategy`. If you are using generic "time window" operations (for example `KeyedStream.timeWindow()` that change behaviour based on the time characteristic, please use equivalent operations that explicitly specify processing time or event time). 
+
+- Remove deprecated `DataStream#split` [FLINK-19083](https://issues.apache.org/jira/browse/FLINK-19083)
+
+  The `DataStream#split()` operation has been removed after being marked as deprecated for a couple of versions. Please use [Side Outputs]({% link dev/stream/side_output.md %})) instead.
+
+- Remove deprecated `DataStream#fold()` method and all related classes [FLINK-19035](https://issues.apache.org/jira/browse/FLINK-19035)
+
+  The long deprecated `(Windowed)DataStream#fold` was removed in 1.12. Please use other operations such as e.g. `(Windowed)DataStream#reduce` that perform better in distributed systems. 
+
+#### Remove remaining UdfAnalyzer configurations [FLINK-13857](https://issues.apache.org/jira/browse/FLINK-13857)
+
+The `ExecutionConfig#get/setCodeAnalysisMode` method and `SkipCodeAnalysis` class were removed. They took no effect even before that change, therefore there is no need to use any of these. 
+
+#### Extend CompositeTypeSerializerSnapshot to allow composite serializers to signal migration based on outer configuration [FLINK-17520](https://issues.apache.org/jira/browse/FLINK-17520)
+
+The `boolean isOuterSnapshotCompatible(TypeSerializer)` on the `CompositeTypeSerializerSnapshot` class has been deprecated, in favor of a new `OuterSchemaCompatibility#resolveOuterSchemaCompatibility(TypeSerializer)` method. Please implement that instead. Compared to the old method, the new method allows composite serializers to signal state schema migration based on outer schema and configuration. 
+
+#### Bump Scala Macros Version to 2.1.1 [FLINK-19278](https://issues.apache.org/jira/browse/FLINK-19278)
+
+Flink now relies on Scala Macros 2.1.1. This means that we no longer support Scala < 2.11.11. 
+
+#### Allow explicitly configuring time behaviour on CEP PatternStream [FLINK-19326](https://issues.apache.org/jira/browse/FLINK-19326)
```

Comment:
This could be further up there with the same notice for `KeyedStream.intervalJoin()`